llvm/lib/Target/Hexagon/HexagonRegisterInfo.cpp

```diff
@@ -431,8 +431,9 @@ unsigned HexagonRegisterInfo::getHexagonSubRegIndex(
       return WSub[GenIdx];
   }
 
-  if (const TargetRegisterClass *SuperRC = *RC.getSuperClasses())
-    return getHexagonSubRegIndex(*SuperRC, GenIdx);
+  unsigned SuperID = *RC.superclasses().begin();
+  if (SuperID != ~0U)
```

Comment:
I don't see how this would work now. Check `superclasses().empty()` instead?

Reply:
Done. Seems unfortunate that no Hexxagon tests expose this...